### DISCUSSION TITLE
Multi-tenant engine

### DISCUSSION THREAD

**Discussion by marcosnils** [dagger/dagger#12069] [Ideas]
### What are you trying to do?

Having two different clients taking to the same Dagger engine to have different views of the API depending on the extensions that each decided to include, currently the engine exposes a single global API for everyone.

### Why is this important to you?

So we can provide multi-tenancy clients in the same engine

### How are you currently working around this?

N/A

cc @sipsma @aluzzardi

**Comment by sipsma** [comment]
Related to this issue, which is essentially the same but specific to extensions (whereas this new one is about clients): https://github.com/dagger/dagger/issues/3072
